### PR TITLE
[hotfix] Added condition to Poa limit reached to avoid spam 

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

   A few sentences describing the overall goals of the pull request's commits.

   Explain any technical decisions that were made and why (when it makes sense).

**Database changed?**

   Yes | No

**Where should the reviewer start?**

- [ ] Foo.java
- [ ] Foo.kt

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APPC-XXXX](https://aptoide.atlassian.net/browse/APPC-<Jira ticket number>)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-XXXX](https://aptoide.atlassian.net/browse/APPC-<Jira ticket number>)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/advertise/CampaignInteract.kt [hunk 1]
```diff
@@ -90,4 +93,24 @@ class CampaignInteract(private val campaignService: CampaignService,
     return campaignService.retrievePoaInformation(address)
   }
 
+  /**
+   * Checks if the user has seen the Poa notification in the last 12h
+   */
+  override fun hasSeenPoaNotification(): Boolean {
+    val savedTime = sharedPreferenceRepository.getPoaNotifiationSeenTime()
+    if (savedTime != (-1).toLong()) {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/advertise/CampaignInteract.kt:101]: There's no need to use "(-1).toLong()" , 0L should be enough

File: app/src/main/java/com/asfoundation/wallet/repository/PreferenceRepositoryType.kt [hunk 2]
```diff
@@ -11,4 +11,7 @@ interface PreferenceRepositoryType {
   fun isFirstTimeOnTransactionActivity(): Boolean
   fun setFirstTimeOnTransactionActivity()
 
+  fun setPoaNotificationSeenTime(currentTimeInMillis: Long)
+  fun getPoaNotifiationSeenTime(): Long
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/repository/PreferenceRepositoryType.kt:15]: Small typo in word Notification

File: app/src/main/java/com/asfoundation/wallet/repository/PreferenceRepositoryType.kt [hunk 3]
```diff
@@ -11,4 +11,7 @@ interface PreferenceRepositoryType {
   fun isFirstTimeOnTransactionActivity(): Boolean
   fun setFirstTimeOnTransactionActivity()
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/repository/PreferenceRepositoryType.kt:13]: No need for this empty line

File: app/src/main/java/com/asfoundation/wallet/advertise/Advertising.kt [hunk 4]
```diff
@@ -16,6 +16,10 @@ interface Advertising {
                         versionCode: Int): Single<ProofSubmissionData>
 
   fun retrievePoaInformation(address: String): Single<PoaInformationModel>
+
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/advertise/Advertising.kt:19]: No need for this empty line

File: app/src/main/java/com/asfoundation/wallet/advertise/AppCoinsAdvertisingBinder.kt [hunk 5]
```diff
@@ -32,7 +32,16 @@ internal class AppCoinsAdvertisingBinder(
...
+        .doOnSuccess {
+          if (it.hasReachedPoaLimit()) {
+            if (!campaignInteract.hasSeenPoaNotification()) {
+              showNotification(it, pkgInfo)
+              campaignInteract.saveSeenPoaNotification()
+            }
+          } else {
+            campaignInteract.clearSeenPoaNotification()
+          }
+        }
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/advertise/AppCoinsAdvertisingBinder.kt]: I think that it's better if we extract this logic into a private method to improve readability.
Extra: why negate the if condition if we can change the method name?

File: app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java [hunk 6]
```diff
@@ -191,9 +192,13 @@ private void processWalletState(ProofSubmissionData proof, Intent intent, String
         //No campaign or already rewarded so there is no need to notify the user of anything
         proofOfAttentionService.remove(packageName);
         if (proof.hasReachedPoaLimit()) {
-          showNotification(proof);
-          stopForeground(false);
+          if (!campaignInteract.hasSeenPoaNotification()) {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java]: Extra: why negate the if condition if we can change the method name?

**Review by carvalho2707 — CHANGES_REQUESTED** [review]

**Comment by JDSM01** [comment]
This was not based on master and so another PR will be opened

**Pull request closed without merging**